gazebo_ros/src/gazebo_ros_effort.cpp

```diff
@@ -0,0 +1,451 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gazebo/physics/Link.hh>
+#include <gazebo/physics/Model.hh>
+#include <gazebo/physics/PhysicsIface.hh>
+#include <gazebo/physics/World.hh>
+#include <gazebo_msgs/srv/apply_body_wrench.hpp>
+#include <gazebo_msgs/srv/body_request.hpp>
+#include <gazebo_msgs/srv/apply_joint_effort.hpp>
+#include <gazebo_msgs/srv/joint_request.hpp>
+#include <gazebo_ros/node.hpp>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "gazebo_ros/conversions/builtin_interfaces.hpp"
+#include "gazebo_ros/gazebo_ros_effort.hpp"
+
+namespace gazebo_ros
+{
+class GazeboRosEffortPrivate
+{
+public:
+  /// Apply body wrench task
+  struct BodyWrenchTask
```

Comment:
I forgot to mention that we should be changing references to "body" to say "link" instead, because body is not a proper term in the Gazebo / SDF spec. Since this PR is already implemented, we can leave it as is, but this is something to keep in mind for the future. I've also added a [note](https://github.com/ros-simulation/gazebo_ros_pkgs/wiki/ROS-2-migration-contribution-guide/_compare/f7b8745fc45e4c2e68a62c208ad9cbc0ff753c15...6d544c7c24133cab22d7d17dff7f4273cdb5a6b1) about this to the migration guide.

Reply:
I will make the changes in this as well. Also do you have any suggestions for a better name for the plugin itself? `gazebo_ros_effort` seems to be indicating only the `joint_efforts` and not the wrenches. I thought of `gazebo_ros_force` but there already exists one in `gazebo_plugins`

Reply:
Yeah I agree that naming is complicated here. Maybe `gazebo_ros_apply_force`? or `gazebo_ros_force_system`, since it's a system plugin?